flytekit/image_spec/image_spec.py

```diff
@@ -106,7 +108,7 @@ def exist(self) -> bool:
             return True
 
     def __hash__(self):
-        return hash(self.to_json())
+        return hash(asdict(self).__str__())
```

Comment:
why do we need this?

Reply:
I removed the dataclass_json decorator from imageSpec so that we can see [postfix code completion﻿](https://www.jetbrains.com/help/pycharm/auto-completing-code.html#postfix_completion) when creating imageSpec in the IDE